network/network_windows.go

```diff
@@ -318,6 +325,47 @@ func (nm *networkManager) configureHcnNetwork(nwInfo *NetworkInfo, extIf *extern
 	return hcnNetwork, nil
 }
 
+func (nm *networkManager) addIPv6DefaultRoute() error {
+	// add ipv6 default route if it does not exist in dualstack overlay windows node from persistentstore
+	cmd := `((Get-NetIPInterface | where InterfaceAlias  -Like "vEthernet*").IfIndex)[0]`
+	ifIndex, err := nm.plClient.ExecutePowershellCommand(cmd)
+	if err != nil {
+		return errors.Wrapf(err, "error while executing powershell command to get ipv6 Hyper-V interface")
+	}
+
+	getIPv6RouteActiveCmd := fmt.Sprintf("Get-NetRoute -DestinationPrefix %s", defaultIPv6Route)
+	getIPv6RoutePersistentCmd := fmt.Sprintf("Get-NetRoute -DestinationPrefix %s -PolicyStore Persistentstore", defaultIPv6Route)
+	if out, err := nm.plClient.ExecutePowershellCommand(getIPv6RoutePersistentCmd); err != nil {
+		logger.Info("ipv6 default route is not found from persistentstore, adding default ipv6 route to the windows node", zap.Any("out", out))
+		// run powershell cmd to add ipv6 default route
+		addCmd := fmt.Sprintf("Remove-NetRoute -DestinationPrefix %s -InterfaceIndex %s -NextHop %s -confirm:$false;New-NetRoute -DestinationPrefix %s -InterfaceIndex %s -NextHop %s -confirm:$false",
+			defaultIPv6Route, ifIndex, defaultIPv6NextHop, defaultIPv6Route, ifIndex, defaultIPv6NextHop)
+
+		// if command is failed to execute, then attempt to execute extra two times
+		for i := 0; i <= addIPv6DefaultRouteAttempts; i++ {
+			if out, err := nm.plClient.ExecutePowershellCommand(addCmd); err != nil {
+				logger.Error("Failed to add ipv6 default gateway route, retrying after error", zap.Any("out", out), zap.Error(err))
+			} else {
+				// check if ipv6 default route is added to both active and persistent store
```

Comment:
whats difference between active and persistent store? why it has to be in both store?

Reply:
By default, it gets added to both, but also you can't just store something in the persistent store without it being in the active store: https://learn.microsoft.com/en-us/powershell/module/nettcpip/new-netroute?view=windowsserver2022-ps#-policystore

Reply:
can it be only in active store? on reboot, cni has to recreate both hns network and this route.. usually everything should come clean after reboot

Reply:
Prince confirmed that if cni is recreating the network on every reboot, you can keep the route only in activestore. And no need to keep in persistentstore

Reply:
yes lets do that